src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1434,83 +1437,67 @@ def aks_use_dev_spaces(cmd, client, cluster_name, resource_group_name, space_nam
 
 
-def aks_remove_dev_spaces(cmd, client, cluster_name, resource_group_name, prompt=False):  # pylint: disable=line-too-long
+def aks_remove_dev_spaces(cmd, client, name, resource_group_name, prompt=False):
     """
     Remove Azure Dev Spaces from a managed Kubernetes cluster.
 
-    :param cluster_name: Name of the managed cluster.
-    :type cluster_name: String
+    :param name: Name of the managed cluster.
+    :type name: String
     :param resource_group_name: Name of resource group. You can configure the default group. \
     Using 'az configure --defaults group=<name>'.
     :type resource_group_name: String
     :param prompt: Do not prompt for confirmation.
     :type prompt: bool
     """
 
-    dev_spaces_extension = 'dev-spaces-preview'
-
-    should_delete_resource = False
-    if not _is_dev_spaces_extension_installed(dev_spaces_extension):
-        if _install_dev_spaces_extension(dev_spaces_extension):
-            should_delete_resource = True
-    else:
-        should_delete_resource = True
-
-    if should_delete_resource:
+    if _is_dev_spaces_extension_installed(dev_spaces_extension):
+        azext_custom = _get_azext_module(dev_spaces_extension)
         try:
-            # Adding the installed extension in the path
-            from azure.cli.core.extension import get_extension_path
-            ext_dir = get_extension_path(dev_spaces_extension)
-            sys.path.append(ext_dir)
-            # Import the extension module
-            from importlib import import_module
-            azext_custom = import_module('azext_dev_spaces_preview.custom')
             azext_custom.ads_remove_dev_spaces(
-                cluster_name, resource_group_name, prompt)
-        except Exception as e:  # nopa pylint: disable=broad-except
-            raise CLIError(e)
+                name, resource_group_name, prompt)
+        except AttributeError as ae:
+            raise CLIError(ae)
 
 
-def _install_dev_spaces_extension(dev_spaces_extension):
+def _get_azext_module(extension_name):
+    try:
+        # Adding the installed extension in the path
+        from azure.cli.core.extension import get_extension_path
+        ext_dir = get_extension_path(extension_name)
+        sys.path.append(ext_dir)
+        # Import the extension module
+        from importlib import import_module
+        azext_custom = import_module('azext_dev_spaces_preview.custom')
+        return azext_custom
+    except ImportError as ie:
+        raise CLIError(ie)
+
+
+def _install_dev_spaces_extension(extension_name):
     try:
         from azure.cli.command_modules.extension import custom
-        custom.add_extension(extension_name=dev_spaces_extension)
+        custom.add_extension(extension_name=extension_name)
     except Exception:  # nopa pylint: disable=broad-except
         return False
     return True
 
 
-def _is_dev_spaces_extension_installed(dev_spaces_extension):
+def _is_dev_spaces_extension_installed(extension_name):
```

Comment:
Undesirable function name. The name indicates this method is a read-only getter. However, the method actually installs extension when the extension is missing. A better name will look like `get_or_add_extension`

Reply:
Will go with _get_or_add_extension()